app/controllers/assignments_controller.rb

```diff
@@ -28,26 +30,47 @@ def create
   end
 
   # rubocop:disable MethodLength
-  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable AbcSize
   def show
     @assignment_repos = AssignmentRepo
       .where(assignment: @assignment)
       .order(:id)
       .page(params[:page])
+
     return unless @organization.roster
 
     @roster_entries = @organization.roster.roster_entries
-      .order(:id)
       .page(params[:students_page])
       .order_for_view(@assignment)
+      .order_by_sort_mode(@current_sort_mode, assignment: @assignment)
 
     @unlinked_user_repos = AssignmentRepo
       .where(assignment: @assignment, user: @unlinked_users)
       .order(:id)
       .page(params[:unlinked_accounts_page])
   end
+  # rubocop:enable AbcSize
+  # rubocop:enable MethodLength
+
+  # rubocop:disable MethodLength
+  def filter_repos
+    return unless @organization.roster
+
+    matching_roster_entries = @organization.roster.roster_entries
+    if @query.present?
+      matching_roster_entries = matching_roster_entries.where("identifier LIKE ?", "%#{@query}%")
+    end
+
+    @roster_entries = matching_roster_entries
+      .page(params[:students_page])
```

Comment:
How does this work when we're not on the first page? It'd be weird to be paginating by created_at, then apply a search and be given results on the nth page. Or what if we're on the second page, and the results from search don't fill a full page? Will this render an empty set?

We probably want to reset the page param when we perform a search.

Reply:
You're right, good catch